actors/evm/src/lib.rs

```diff
@@ -118,14 +119,23 @@ impl EvmContractActor {
 
         // TODO this does not return revert data yet, but it has correct semantics.
         if exec_status.reverted {
-            Err(ActorError::unchecked(EVM_CONTRACT_REVERTED, "constructor reverted".to_string()))
+            rt.exit(
+                EVM_CONTRACT_REVERTED.value(),
+                RawBytes::from(exec_status.output_data.to_vec()),
+                Some("constructor reverted"),
+            )
         } else if exec_status.status_code == StatusCode::Success {
             system.set_bytecode(&exec_status.output_data)?;
             system.flush()
         } else if let StatusCode::ActorError(e) = exec_status.status_code {
             Err(e)
         } else {
-            Err(ActorError::unspecified("EVM constructor failed".to_string()))
+            let msg = format!("{}", exec_status.status_code);
+            rt.exit(
+                EVM_CONTRACT_EXECUTION_ERROR.value(),
+                RawBytes::from(msg.as_bytes().to_vec()),
+                Some("constructor failed"),
+            )
```

Comment:
Ok, I take back what I said before. We absolutely can't cook up this value. The return data is user-specified and sacred.

Reply:
Wait, there will be no user data and no bactrace here whatsoever.
If we dont do this, it will be a "See Figure 1" type of error.

Reply:
ok, wait, we can put the message in the backtrace I guess. Fine, fine, I will observe your sacred cow and not synthesize the error message.